src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/LineChart/SubscaleLineChart/ChartTooltip/ChartTooltip.test.tsx

```diff
@@ -36,31 +38,56 @@ describe('ChartTooltip', () => {
     const tooltip = screen.queryByTestId(`${dataTestid}-tooltip`);
-    expect(tooltip).not.toBeInTheDocument();
+    expect(tooltip).toBeNull(); // Using Jest's default matcher to check if the element is not present
   });
 
   test('renders component correctly', () => {
     render(<ChartTooltip {...props} />);
     const tooltip = screen.getByTestId(`${dataTestid}-tooltip`);
-    expect(tooltip).toBeInTheDocument();
+    expect(tooltip).not.toBeNull(); // Ensures the tooltip is rendered
 
     const regex = new RegExp(`${dataTestid}-tooltip-item-\\d+$`);
     const tooltipItems = screen.queryAllByTestId(regex);
-    expect(tooltipItems).toHaveLength(2);
+    expect(tooltipItems.length).toBe(2); // Check if there are two items
 
     const itemContainer0 = screen.getByTestId(`${dataTestid}-tooltip-item-0`);
-    expect(itemContainer0).toHaveTextContent('Example 1: 42');
+    expect(itemContainer0.textContent).toContain('Example 1: 42');
     const mdEditor0 = within(itemContainer0).getByTestId(`${dataTestid}-tooltip-item-0-md-preview`);
-    expect(mdEditor0).toBeInTheDocument();
-    expect(itemContainer0).toHaveTextContent('Mocked Option Text');
-    expect(itemContainer0).toHaveTextContent('Dec 20, 14:55');
+    expect(mdEditor0).not.toBeNull();
+    expect(itemContainer0.textContent).toContain('Mocked Option Text');
+    expect(itemContainer0.textContent).toContain('Dec 20, 14:55');
 
     const itemContainer1 = screen.getByTestId(`${dataTestid}-tooltip-item-1`);
-    expect(itemContainer1).toHaveTextContent('Example 2: 7');
+    expect(itemContainer1.textContent).toContain('Example 2: 7');
     const mdEditor1 = within(itemContainer1).queryByTestId(
       `${dataTestid}-tooltip-item-1-md-preview`,
     );
-    expect(mdEditor1).not.toBeInTheDocument();
-    expect(itemContainer1).toHaveTextContent('Dec 20, 17:20');
+    expect(mdEditor1).toBeNull();
+    expect(itemContainer1.textContent).toContain('Dec 20, 17:20');
+  });
+
+  test('displays tooltip on hover and verifies formatting', () => {
+    render(<ChartTooltip {...props} />);
+
+    const dataPoint = screen.getByText('Example 1: 42');
+    fireEvent.mouseOver(dataPoint);
+
+    const tooltip = screen.getByTestId(`${dataTestid}-tooltip`);
+    expect(tooltip).not.toBeNull(); // Check tooltip is rendered
+    expect(tooltip).toHaveTextContent('Example 1: 42');
+    expect(tooltip).toHaveTextContent('Dec 20, 14:55');
+
+    const styles = window.getComputedStyle(tooltip);
+
+    expect(styles.minWidth).toBe('38rem');
+    expect(styles.maxWidth).toBe('62rem');
+    expect(styles.overflowY).toBe('auto');
+    expect(styles.padding).toBe('20px 20px 0px 20px');
+    expect(styles.margin).toBe('0px 16px 0px 0px');
+    expect(styles.borderRadius).toBe('1.2rem');
+
+    const optionalTextElement = within(tooltip).getByText('Mocked Option Text');
+    expect(optionalTextElement).toBeInTheDocument();
+    //const optionalTextStyles = window.getComputedStyle(optionalTextElement);
```

Comment:
I'm not sure if this test is useful. Should we be validating styles like this in a unit test? It feels like too much implementation detail is in the test and it would be better served just manually observing it

Reply:
+1 to Kenroy's comment – we generally don't check styling in unit tests (at least on this project), even when a solution implementation is styling-related. That's generally handled by peer testing and FQA/VQA.

Reply:
Thanks, I wasn't sure If I should include them. Also I was thinking in what kind of tests I should include for this ticket. I deleted them on the last PR.